### PR TITLE
CLN: make kwargs explicit in pytables constructors

### PR DESCRIPTION
**Pull request by jbrockmendel**
Orthogonal to #29935.

### CODE REVIEW

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -3303,18 +3307,18 @@ def queryables(self) -> Dict[str, Any]:
...
+        d1 = [(a.cname, a) for a in self.index_axes]
+        d2 = [
+            (self.storage_obj_type._AXIS_NAMES[axis], None)
+            for axis, values in self.non_index_axes
+        ]
+        d3 = [
+            (v.cname, v) for v in self.values_axes if v.name in set(self.data_columns)
+        ]
+
+        return dict(d1 + d2 + d3)  # type: ignore
```
Comment (WillAyd) [inline pandas/io/pytables.py]: Could probably ignore this if you don’t try to add the lists together and just update the Dict as required

**Review by WillAyd — APPROVED** [review]
Lgtm. Optional comment

File: pandas/io/pytables.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/io/pytables.py]: will give this a shot

Comment (jbrockmendel) [reply pandas/io/pytables.py]: after a couple tries at this, its landing in "more trouble than its worth" territory

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**